spec/ffi/function_type_spec.rb

```diff
@@ -0,0 +1,27 @@
+#
+# This file is part of ruby-ffi.
+# For licensing, see LICENSE.SPECS
+#
+
+require File.expand_path(File.join(File.dirname(__FILE__), "spec_helper"))
+
+describe FFI::FunctionType do
+  it 'is initialized with return type and a list of parameter types' do
+    function_type = FFI::FunctionType.new(:int, [ :char, :size_t ])
+    expect(function_type.result_type).to be == FFI::Type::Builtin::INT
+    expect(function_type.param_types).to be == [ FFI::Type::Builtin::CHAR, FFI::Type::Builtin::ULONG ]
+  end
+
+  it 'has a memsize function' do
+    base_size = ObjectSpace.memsize_of(Object.new)
+
+    function_type = FFI::FunctionType.new(:int, [])
+    size = ObjectSpace.memsize_of(function_type)
+    expect(size).to be > base_size
```

Comment:
The size can vary on platforms etc, so I usually use this pattern to ensure the function report something, but not assert a precise size.